lib/xenops_server.ml

```diff
@@ -2572,8 +2572,14 @@ module VM = struct
       (fun () ->
          let vgpu_id =
            (* The URI is /service/xenops/migrate-vgpu/id *)
-           let bits = Stdext.Xstringext.String.split '/' (Uri.path uri) in
-           let vgpu_id_str = bits |> List.rev |> List.hd in
+           let path = Uri.path uri in
+           let bits = Astring.String.cut ~sep:"/" ~rev:true path in
```

Comment:
The code is only interested in the leaf of the tree, `cut` with `~rev:true` accomplishes this.